### PR TITLE
chore/eslint return type

### PR DESCRIPTION
**Pull request by tobiasqueck**
* adding rule `"@typescript-eslint/explicit-function-return-type": "warn"` (only `src`, disabled for `test` folders)
* as example fix all new warnings in `@sap-ux/create`

### CODE REVIEW

**Review by johannes-kolbe — APPROVED** [review]
- Configuration for code and tests lg
- Updates to `create` package lg
- Changeset provided
- Did not test manually

File: packages/create/src/tracing/logger.ts [hunk 1]
```diff
@@ -35,8 +35,8 @@ function setCustomFormatter(logger: ToolsLogger): void {
     }
     const consoleTransport = transports.find((t) => t?.name === 'console');
     if (consoleTransport?.format) {
-        consoleTransport.format.transform = (info: any) => {
-            const colorFn = levelColor[info.level] ? chalk.keyword(levelColor[info.level]) : (m: string) => m;
+        consoleTransport.format.transform = (info: any): any => {
```
[the remark this answers is not in the document]
Comment (tobiasqueck) [reply packages/create/src/tracing/logger.ts:38]: This function is terrible and needs fixing, however, I couldn't get it fixed without spending more time on it. The main problem is that it uses private variables from the logger module. So, an ideal fix should expose what is needed there first (but that doesn't belong into this PR).

**Review by heimwege — APPROVED** [review]
lgtm

code changes ok
changeset ok
did NOT test manually

**Review by devinea — APPROVED** [review]
Adding eslint rule looks good. 
changeset ✅

**Pull request merged**